README.md

```diff
@@ -3,3 +3,32 @@
 This extension is Che specific and works only inside Che workspace.
 
 The main purpose of this extension is to take care about url changing and/or port forwarding caused by running Theia inside container.
+
+### Technical info
+
+#### What is the reason for this extension?
+
+This extension is designed to provide correct route for end user (and main Theia frontend) when Theia runs inside Eclipse Che.
+
+#### When this extension is used?
+
+Each time when an extra Theia instance is needed.
+For now, it is used for hosted plugin instance.
+
+#### Why it doesn't work without this extension?
+
+Theia server starts on some port.
+If Theia runs natively a user can just use direct url, but in case when Theia is running inside Eclipse Che workspace direct url won't work. There should be a dedicated server in workspace config to expose needed port and this server has a preview url which is designed to access the resource from outside.
+
+#### How does this extension work?
+
+It retrieves hosted instance url form workspace config.
+Supposed, that hosted instance server has attribute "type" with "ide-dev" value.
+
+#### How to test the extension
+
+Just start hosted plugin instance (use `Hosted Plugin: Start Instance` command). If a new browser tab with hosted instance is opened then it works as expected.
```

Comment:
Maybe provide a link to a sample plugin to be used in Github or how to create it quickly.